### PR TITLE
Installation via pip

### PR DESCRIPTION
**Pull request by ir5**
Update: Now ready for review

This PR will enable to install chianer-compiler via pip.
TODO:

- [x] Fix hard-coded paths
- [x] Fix examples
- ~Update requirements to install the appropriate version of chainer~
- [x] Catch up the latest master branch

One concern:
Current code under `python` and `example` directories is assuming that chainer_compiler is installed via `setup.py`. However, CI will run without the installation via `setup.py`.
So, in test code, I put a little complicated code to modify path configuration. We may improve these parts in the future.
Also, we will have to  add documentation about the installation via pip.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            '-DCHAINER_COMPILER_ENABLE_CUDA=ON',
+            '-DCHAINERX_BUILD_CUDA=ON',
+            '-DCUDA_TOOLKIT_ROOT_DIR=' + cuda_path,
+            '-DCHAINER_COMPILER_ENABLE_PYTHON=ON',
+            '-DPYTHON_EXECUTABLE=' + python3_executable,
+            '-DCHAINERX_BUILD_PYTHON=ON',
+            '-DCUDNN_ROOT_DIR=' + cudnn_root_dir,
+        ]
+        build_args = [
+            '--', '-j4',
```
Comment (shinh) [inline setup.py]: I think this can kill machines with small RAMs. Shouldn't we just rely on `MAKEFLAGS` set outside this script?

Comment (ir5) [reply setup.py]: I agree with you. I would like to add `MAKEFLAGS` as an optional flag.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        if cudnn_root_dir is None:
+            raise RuntimeError('CUDNN_ROOT_DIR must be set.')
+
+        cmake_args = [
+            '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
+            '-DCHAINER_COMPILER_ENABLE_CUDA=ON',
+            '-DCHAINERX_BUILD_CUDA=ON',
+            '-DCUDA_TOOLKIT_ROOT_DIR=' + cuda_path,
+            '-DCHAINER_COMPILER_ENABLE_PYTHON=ON',
+            '-DPYTHON_EXECUTABLE=' + python3_executable,
```
Comment (disktnk) [inline setup.py]: `sys.executable` returns python path

File: python/chainer_compiler/chainer_compiler.py [hunk 3]
```diff
@@ -169,6 +179,7 @@ def backward(self, indexes, flat_gys):
 
 def _run_translator(translator, mc, inputs):
     if translator == 'ch2o':
+        import ch2o
```
[the remark this answers is not in the document]
Comment (ir5) [reply python/chainer_compiler/chainer_compiler.py]: Yes, we should consider how to import ch2o and elichika. I want to leave this task as future work, though.

File: setup.py (hunk 2 above)
Comment (ir5) [reply setup.py]: Thank you, I modified the code.

**Comment by shinh** [comment]
LGTM, please merge this if this PR is ready.

**Comment by ir5** [comment]
Thank you, I would merge.

**Pull request merged**